### PR TITLE
feat(sidebar): enhancements

### PR DESCRIPTION
**Pull request by daanishnasir**
- Can add footer nav item capability and will collapse correctly
- Add avatar and avatar props to sidebar-nav-item
- Add title, subtitle props
- fix icon style
- Remove expanded css class as was not used; confirmed.
- Add navitem class prop
- Header will be hidden on minimized sidebar
- Added momentum logo to assets 

## Screenshots:
**Before** (If applicable):

**After**:

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: angular/src/lib/sidebar-nav-item/sidebar-nav-item.component.ts [hunk 1]
```diff
@@ -50,18 +51,38 @@ import { SidebarService } from '../sidebar/sidebar.service';
...
-        ]" style="font-size: 20px;"></i>
+          icon ? 'icon-' + this.icon : '',
+          iconClass
+        ]"></i>
       </div>
 
-      <div class="md-list-item__center">{{ title }}</div>
+      <ng-container *ngIf="avatarTitle && !icon">
+        <div class="md-list-item__left">
+          <md-avatar
```
Comment (bfbiggs) [inline angular/src/lib/sidebar-nav-item/sidebar-nav-item.component.ts]: @pauljeter daani and I looked at this yesterday and didn't know if the footer avatar should be considered as a sidebar-nav-item

File: core/scss/components/sidebar/sidebar.scss [hunk 2]
```diff
@@ -6,7 +6,7 @@
   .#{$sidebar-class} {
     $parent: &;
     z-index: $zindex-sidebar;
-    height: 100%;
```
Comment (bfbiggs) [inline core/scss/components/sidebar/sidebar.scss]: Let's discuss, was running into issues using this method. I think we could do min-height: 100vh and height: 100% to cover both uses.

File: core/scss/components/sidebar/sidebar.scss [hunk 3]
```diff
@@ -49,6 +52,15 @@
     &--minimized {
       width: rem-calc(68);
       min-width: 0;
+
+      .md-avatar__img {
```
Comment (bfbiggs) [inline core/scss/components/sidebar/sidebar.scss]: Let's discuss this - should this be a class applied to avatar like sidebar__footer--avatar or target avatar? If targeting avatar is the best solution, then let's use a variable here.

Comment (daanishnasir) [reply core/scss/components/sidebar/sidebar.scss]: its changed to md-sidebar__avatar

File: core/scss/components/sidebar/sidebar.scss (hunk 2 above)
Comment (daanishnasir) [reply core/scss/components/sidebar/sidebar.scss]: switched to min-height and height

File: angular/src/lib/sidebar-nav-item/sidebar-nav-item.component.ts (hunk 1 above)
Comment (daanishnasir) [reply angular/src/lib/sidebar-nav-item/sidebar-nav-item.component.ts]: Avatar taken out of sidebar code.

File: angular/src/docs/app/playground/playground.component.ts [hunk 4]
```diff
@@ -15,4 +16,5 @@ export class PlaygroundComponent implements OnInit {
   onSelect (event) {
     console.info('custom onSelect working');
   }
+  // <example-popover-default></example-popover-default>
```
Comment (bfbiggs) [inline angular/src/docs/app/playground/playground.component.ts]: Should these changes be discarded?

File: angular/src/lib/sidebar-nav-item/sidebar-nav-item.component.ts [hunk 5]
```diff
@@ -83,6 +84,8 @@ import { SidebarService } from '../sidebar/sidebar.service';
 export class SidebarNavItemComponent implements AfterContentInit, OnInit, OnDestroy {
   /** @prop Title for the side item navigation | '' */
   @Input() title: string = '';
+  /** @prop Sub title string  */
+  @Input() subtitle: string = '';
```
Comment (bfbiggs) [inline angular/src/lib/sidebar-nav-item/sidebar-nav-item.component.ts]: Are you using this?

File: angular/src/lib/sidebar-nav-item/sidebar-nav-item.component.ts [hunk 6]
```diff
@@ -97,6 +100,10 @@ export class SidebarNavItemComponent implements AfterContentInit, OnInit, OnDest
   @Input() isReadOnly: boolean = false;
   /** @prop creates a separator line below the sidebar nav item */
   @Input() separator: boolean = false;
+  /** @prop optional CSS class for the icon left of sidebar nav item title */
+  @Input() iconClass: string = '';
+  /** @prop optional CSS class on the sidebar nav item wrapper div */
+  @Input() navItemClass: string = '';
```
Comment (bfbiggs) [inline angular/src/lib/sidebar-nav-item/sidebar-nav-item.component.ts]: I've seen className for the core component in other components...what's the pattern we're following for the main component?

File: angular/src/lib/sidebar/examples/sidebar-pageLevel.component.ts [hunk 7]
```diff
@@ -4,18 +4,17 @@ import { Component } from '@angular/core';
   selector: 'example-sidebar-page-level',
```
Comment (bfbiggs) [inline angular/src/lib/sidebar/examples/sidebar-pageLevel.component.ts]: @pauljeter should these example components be camelCase or with kebab? Referring to the actual file name not the selector.

File: angular/src/lib/sidebar/sidebar.component.ts [hunk 8]
```diff
@@ -56,8 +56,8 @@ export class SideBarComponent implements OnInit, OnDestroy {
   @Input() isPageLevel = false;
   /** @prop Sets padding for Topbar | false */
   @Input() withTopbar = false;
-  /** @prop Changes padding based on Icon nav | true */
-  @Input() withIcons = true;
+  /** @prop show minimized sidebar with icons only on toggle | true */
```
Comment (bfbiggs) [inline angular/src/lib/sidebar/sidebar.component.ts]: Prop naming wise should this be more generic/explicit like shouldMinimize vs iconNav? @pauljeter

File: core/scss/components/sidebar/sidebar.scss [hunk 9]
```diff
@@ -49,6 +52,20 @@
     &--minimized {
       width: rem-calc(68);
       min-width: 0;
+
+      .md-sidebar__avatar {
```
Comment (bfbiggs) [inline core/scss/components/sidebar/sidebar.scss]: Did we talk about changing these back to avatar specific?

File: angular/src/lib/sidebar-nav-item/sidebar-nav-item.component.ts (hunk 5 above)
Comment (daanishnasir) [reply angular/src/lib/sidebar-nav-item/sidebar-nav-item.component.ts]: removed as its not included now

File: core/scss/components/sidebar/sidebar.scss (hunk 9 above)
Comment (daanishnasir) [reply core/scss/components/sidebar/sidebar.scss]: taking this out, shouldn't have any avatar in the sidebar styles

File: angular/src/lib/sidebar-nav-item/sidebar-nav-item.component.ts (hunk 6 above)
Comment (daanishnasir) [reply angular/src/lib/sidebar-nav-item/sidebar-nav-item.component.ts]: switching it to className as its the main component.

File: angular/src/lib/sidebar/examples/sidebar-pageLevel.component.ts (hunk 7 above)
Comment (daanishnasir) [reply angular/src/lib/sidebar/examples/sidebar-pageLevel.component.ts]: good catch, file name should also be camel cased

File: angular/src/lib/sidebar/sidebar.component.ts [hunk 10]
```diff
@@ -56,8 +57,8 @@ export class SideBarComponent implements OnInit, OnDestroy {
...
+  @Input() withIcons = false;
```
Comment (bfbiggs) [inline angular/src/lib/sidebar/sidebar.component.ts]: Shouldn't default be true?

File: core/scss/components/sidebar/sidebar.scss [hunk 11]
```diff
@@ -18,10 +18,10 @@
       display: flex;
       flex-direction: row;
       height: 100%;
-    }
 
-    &--expanded {
-      display: block;
+      &--fixed{
```
Comment (bfbiggs) [inline core/scss/components/sidebar/sidebar.scss]: Space after fixed

File: core/scss/components/sidebar/sidebar.scss [hunk 12]
```diff
@@ -32,7 +32,14 @@
       padding-top: $top-bar__height--desktop;
     }
 
-    &__footer,
+    &__footer {
+      width: inherit;
+      display: flex;
+      flex-direction: column;
+      padding: 0;
+      min-height: rem-calc(130);
```
Comment (bfbiggs) [inline core/scss/components/sidebar/sidebar.scss]: Should footer have min-height? Shouldn't it be based on content?

Comment (daanishnasir) [reply core/scss/components/sidebar/sidebar.scss]: padding bottom added instead 1rem

**Pull request merged**